examples/00_Basic_Usage/Implementing_a_custom_Kernel.ipynb

```diff
@@ -0,0 +1,428 @@
+    "# True function is sin(2*pi*x) with Gaussian noise\n",
+    "train_y = torch.sin(train_x * (2 * math.pi)) + torch.randn(train_x.size()) * math.sqrt(0.04)\n",
+    "\n",
+    "# Wrap training, prediction and plotting from the ExactGP-Tutorial into a function, \n",
+    "# so that we do not have to repeat the code later on\n",
+    "def train(model, likelihood, training_iter=training_iter):\n",
+    "    # Use the adam optimizer\n",
+    "    optimizer = torch.optim.Adam(model.parameters(), lr=0.1)  # Includes GaussianLikelihood parameters\n",
+    "\n",
+    "    # \"Loss\" for GPs - the marginal log likelihood\n",
+    "    mll = gpytorch.mlls.ExactMarginalLogLikelihood(likelihood, model)\n",
+    "\n",
+    "    for i in range(training_iter):\n",
+    "        # Zero gradients from previous iteration\n",
+    "        optimizer.zero_grad()\n",
+    "        # Output from model\n",
+    "        output = model(train_x)\n",
+    "        # Calc loss and backprop gradients\n",
+    "        loss = -mll(output, train_y)\n",
+    "        loss.backward()\n",
+    "        optimizer.step()\n",
+    "        \n",
+    "\n",
+    "def predict(model, likelihood, test_x = torch.linspace(0, 1, 51)):\n",
+    "    # Make predictions by feeding model through likelihood\n",
+    "    with torch.no_grad(), gpytorch.settings.fast_pred_var():\n",
+    "        # Test points are regularly spaced along [0,1]\n",
+    "        return likelihood(model(test_x))\n",
+    "\n",
+    "def plot(observed_pred, test_x=torch.linspace(0, 1, 51)):\n",
+    "    with torch.no_grad():\n",
+    "        # Initialize plot\n",
+    "        f, ax = plt.subplots(1, 1, figsize=(4, 3))\n",
+    "\n",
+    "        # Get upper and lower confidence bounds\n",
+    "        lower, upper = observed_pred.confidence_region()\n",
+    "        # Plot training data as black stars\n",
+    "        ax.plot(train_x.numpy(), train_y.numpy(), 'k*')\n",
+    "        # Plot predictive means as blue line\n",
+    "        ax.plot(test_x.numpy(), observed_pred.mean.numpy(), 'b')\n",
+    "        # Shade between the lower and upper confidence bounds\n",
+    "        ax.fill_between(test_x.numpy(), lower.numpy(), upper.numpy(), alpha=0.5)\n",
+    "        ax.set_ylim([-3, 3])\n",
+    "        ax.legend(['Observed Data', 'Mean', 'Confidence'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### A first kernel\n",
+    "\n",
+    "To implement a custom kernel, we derive one from GPyTorch's [`kernel` class](https://docs.gpytorch.ai/en/latest/kernels.html) and implement the `forward()` method. The base class provides many useful routines. For example, `__call__()` is implemented, so that the kernel may be called directly, without resorting to the `forward()` routine. Among other things, the `Kernel` class provides a method `covar_dist()`, which may be used to calculate the Euclidian distance between point pairs conveniently.\n",
```

Comment:
```suggestion
    "To implement a custom kernel, we derive one from GPyTorch's [kernel class](https://docs.gpytorch.ai/en/latest/kernels.html) and implement the `forward()` method. The base class provides many useful routines. For example, `__call__()` is implemented, so that the kernel may be called directly, without resorting to the `forward()` routine. Among other things, the `Kernel` class provides a method `covar_dist()`, which may be used to calculate the Euclidian distance between point pairs conveniently.\n",
```